include/cuco/detail/__config

```diff
@@ -14,9 +14,23 @@
  * limitations under the License.
  */
 
- #pragma once
+#pragma once
 
- #include <nv/target>
+#include <nv/target>
+
+#if !defined(__CUDACC_VER_MAJOR__) || !defined(__CUDACC_VER_MINOR__)
+#error "NVCC version not found"
+#elif __CUDACC_VER_MAJOR__ < 11 || (__CUDACC_VER_MAJOR__ == 11 && __CUDACC_VER_MINOR__ < 5)
+#error "NVCC version 11.6 or later is required"
+#endif
+
+#if !defined(__CUDACC_RELAXED_CONSTEXPR__)
+#error "Support for relaxed constexpr is required"
+#endif
```

Comment:
Valid for now, but one small goal of the refactoring is to get rid of this requirement.